File: medicnote-backend/src/main/java/com/medicnote/backend/service/AuthService.java
package com.medicnote.backend.service;

import com.medicnote.backend.dto.auth.AuthRequest;
import com.medicnote.backend.dto.auth.AuthResponse;
import com.medicnote.backend.dto.auth.RegisterRequest;

public interface AuthService {

    String register(RegisterRequest request);

    AuthResponse login(AuthRequest request);
}

File: medicnote-frontend/src/services/authService.ts
import api from "./api";

/**
 * Base Register Type (common fields)
 */
type BaseRegister = {
  name: string;
  email: string;
  password: string;
  role: "ROLE_PATIENT" | "ROLE_DOCTOR" | "ROLE_ADMIN";
};

/**
 * Doctor Extra Fields
 */
type DoctorExtra = {
  specialization?: string;
  experience?: number;
};

export type RegisterRequest = BaseRegister & DoctorExtra;

export const authService = {
  login: (email: string, password: string) =>
    api.post("/auth/login", {
      email,
      password,
    }),

  register: (data: RegisterRequest) =>
    api.post("/auth/register", data),
};